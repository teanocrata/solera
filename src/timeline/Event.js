import React, { Component } from "react";

const styles = {
  img: {
    width: "100%",
    marginTop: "20px",
    marginBottom: "20px",
    outline: "1px dashed white",
    outlineOffset: "-8px"
  },
  card: {
    background: "lightgreen"
  },
  title: {
    fontFamily: "Shadows Into Light",
    textTransform: "uppercase",
    color: "forestgreen",
    textAlign: "left"
  }
};

class Event extends Component {
  render() {
    return (
      <div style={styles.card} icon={this.props.icon}>
        <h1 style={styles.title}>{this.props.title}</h1>
        <img style={styles.img} alt={this.props.title} src={this.props.img} />
        <p>{this.props.description}</p>
      </div>
    );
  }
}

export default Event;
