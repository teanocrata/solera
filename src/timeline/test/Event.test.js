import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Event from "../Event";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Event />, div);
});

test("Event renders correctly", () => {
  const event = renderer.create(<Event />).toJSON();
  expect(event).toMatchSnapshot();
});
