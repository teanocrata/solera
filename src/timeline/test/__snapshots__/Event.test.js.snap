// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event renders correctly 1`] = `
<div
  icon={undefined}
  style={
    Object {
      "background": "lightgreen",
    }
  }
>
  <h1
    style={
      Object {
        "color": "forestgreen",
        "fontFamily": "Shadows Into Light",
        "textAlign": "left",
        "textTransform": "uppercase",
      }
    }
  />
  <img
    alt={undefined}
    src={undefined}
    style={
      Object {
        "marginBottom": "20px",
        "marginTop": "20px",
        "outline": "1px dashed white",
        "outlineOffset": "-8px",
        "width": "100%",
      }
    }
  />
  <p />
</div>
`;
